File: src/main/java/com/gct/banking_system/entity/Beneficiary.java
package com.gct.banking_system.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "beneficiaries")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Beneficiary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String accountNumber;

    @Column(nullable = false)
    private String bankName;

    @Column(nullable = false)
    private String ifscCode;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    @JsonBackReference
    private Customer customer;
}

File: banking-frontend/src/types/beneficiary.ts
export interface Beneficiary {
  id: number;
  name: string;
  accountNumber: string;
  bankName: string;
  ifscCode: string;
  customerId: number;
  customerName: string;
}

export interface CreateBeneficiaryDTO {
  name: string;
  accountNumber: string;
  bankName: string;
  ifscCode: string;
  customerId: number;
}

export interface UpdateBeneficiaryDTO {
  name: string;
  accountNumber: string;
  bankName: string;
  ifscCode: string;
}
